pkg/frontend/querymiddleware/split_by_instant_interval_test.go

```diff
@@ -153,6 +153,31 @@ func TestQuerySplittingCorrectness(t *testing.T) {
 			query:                `rate(metric_counter[1m]) / rate(metric_counter[5h:5m]) > 0.5`,
 			expectedSplitQueries: 0,
 		},
+		// Offset operator
+		"sum_over_time[3m] offset 3m": {
+			query:                `sum_over_time(metric_counter[3m] offset 3m)`,
+			expectedSplitQueries: 3,
+		},
+		"avg_over_time[3m] offset 5m": {
+			query:                `avg_over_time(metric_counter[3m] offset 5m)`,
+			expectedSplitQueries: 6,
+		},
+		"sum_over_time[3m] offset 30s": {
+			query:                `sum_over_time(metric_counter[3m] offset 30s)`,
+			expectedSplitQueries: 3,
+		},
+		"count_over_time[3m] offset -3m": {
+			query:                `sum_over_time(metric_counter[3m] offset -3m)`,
+			expectedSplitQueries: 3,
+		},
+		"avg_over_time[3m] offset -5m": {
+			query:                `avg_over_time(metric_counter[3m] offset -5m)`,
+			expectedSplitQueries: 3,
+		},
```

Comment:
These tests are failing because we are querying in the future 🪄 
So will probably add some series in the future

Reply:
Either add some series with samples in the future, or reduce the offset to something like `-2m`. Since the time window is `1m` it should work `offset -2m`. Does it?

Reply:
Ah yes, good catch. There is a series that starts 5 minutes from now: https://github.com/grafana/mimir/blob/82ad6be203bacece602f95c59eaa46e1214fbe18/pkg/frontend/querymiddleware/split_by_instant_interval_test.go#L257-L260

Updated and it works 👍